quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/infinispan/CacheManagerFactory.java

```diff
@@ -192,12 +194,16 @@ private static void createRemoteCaches(org.infinispan.client.hotrod.configuratio
         logger.warn("Creating remote cache in external Infinispan server. It should not be used in production!");
         var baseConfig = defaultRemoteCacheBuilder().build();
 
-        Stream.of(USER_SESSION_CACHE_NAME, OFFLINE_USER_SESSION_CACHE_NAME, CLIENT_SESSION_CACHE_NAME, OFFLINE_CLIENT_SESSION_CACHE_NAME, ACTION_TOKEN_CACHE, WORK_CACHE_NAME)
+        Stream.of(ACTION_TOKEN_CACHE, WORK_CACHE_NAME)
                 .forEach(name -> builder.remoteCache(name).configuration(baseConfig.toStringConfiguration(name)));
 
         // We need indexed caches because the delete statement fails for non-indexed cache.
         createIndexedRemoteCache(builder, LOGIN_FAILURE_CACHE_NAME, List.of(LoginFailureEntity.class));
         createIndexedRemoteCache(builder, AUTHENTICATION_SESSIONS_CACHE_NAME, List.of(RootAuthenticationSessionEntity.class));
+        createIndexedRemoteCache(builder, USER_SESSION_CACHE_NAME, List.of(RemoteUserSessionEntity.class));
```

Comment:
Looking at the configurations on KCB, I don't see a hint in the cache configurations that they should be indexed. Is that optional, but recommended? Looking at the comment "We need indexed caches because the delete statement fails for non-indexed cache." is seems that the comment might be outdated, or at least it confuses me as I don't see indexes on the KCB Infinispan scripts.

Reply:
The comment refers to https://issues.redhat.com/browse/ISPN-16270

We updated Infinspan to 15.0.7.Final this week and it makes the comment invalid. I'll revert these changes and create basic and non-indexed caches.

Note that the integration test suite uses this method and is not supposed to be used in production. It is useful for demos since we can start the Infinispan server with the default config :+1: 

For Keycloak 26, indexing is optional but I wouldn't recommend it. Only removing realms uses queries to remove the authentication sessions and login failures. I don't expect removing realms to be such a common operation or being performance sensitive. Also, I don't expect these 2 caches to have a large data set.

Reply:
I've created https://github.com/pruivo/keycloak/pull/4 to clarify the docs about which version of Infinispan to use. Also consider mentioning this in the migration guide. 

I know this method is used in the integration tests, still a production environment shouldn't deviate much from it. And this is something we could automate, not necessarily in this issue though. 

Looking at the code, I see that we will retrieve the client sessions connected to user sessions in several occasions during login / logout, possibly also during token validation. So I'd say "Only removing realms uses queries to remove the authentication sessions and login failures." is only a partial answer to my question.

Reply:
You have answered the the question about the indexing of for the client sessions cache below and that part is now resolved. 

This then leaves only the documentation change PR plus the thing about the non-indexed basic caches open.